### PR TITLE
[Rule Tuning] Merge and Delete duplicate rules for Registration Utilities

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/1027

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/execution_via_net_com_assemblies.toml [hunk 1]
```diff
@@ -1,54 +0,0 @@
-[metadata]
```
[the remark this answers is not in the document]
Comment (brokensound77) [reply rules/windows/execution_via_net_com_assemblies.toml:1]: I just created a PR for this, ref: #1050

File: rules/windows/execution_register_server_program_connecting_to_the_internet.toml [hunk 2]
```diff
@@ -28,12 +28,15 @@ type = "eql"
...
-                  process.name : "RegAsm.exe" or process.name : "RegSvcs.exe") and
-     event.type == "start"]
-  [network where (process.name : "regsvr32.exe" or process.name : "regsvr64.exe" or
-                  process.name : "RegAsm.exe" or process.name : "RegSvcs.exe") and
-     not cidrmatch(destination.ip, "10.0.0.0/8", "169.254.169.254", "172.16.0.0/12", "192.168.0.0/16")]
+  [process where event.type == "start" and
+   process.name : ("regsvr32.exe", "RegAsm.exe", "RegSvcs.exe") and
+   not (
+         user.id == "S-1-5-18" and
+         (process.parent.name : "msiexec.exe" or process.parent.executable:("C:\\Program Files (x86)\\*.exe", "C:\\Program Files\\*.exe"))
```
Comment (brokensound77) [inline rules/windows/execution_register_server_program_connecting_to_the_internet.toml:35]:
```suggestion
         (process.parent.name : "msiexec.exe" or process.parent.executable : ("C:\\Program Files (x86)\\*.exe", "C:\\Program Files\\*.exe"))
```

**Review by brokensound77 — CHANGES_REQUESTED** [review]

File: rules/windows/execution_via_net_com_assemblies.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (brokensound77) [reply rules/windows/execution_via_net_com_assemblies.toml:1]: restore this file and run `python -m detection_rules dev deprecate-rule <rule-path>`.

[the remark this answers is not in the document]
Comment (Samirbous) [reply rules/windows/execution_via_net_com_assemblies.toml:1]: nice, done.

**Review by threat-punter — APPROVED** [review]

**Review by brokensound77 — APPROVED** [review]
Thanks for making those changes

**Pull request merged**